libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -899,3 +951,430 @@ float AP_GPS::get_lag(uint8_t instance) const
+    for (uint8_t i=0; i<GPS_MAX_RECEIVERS; i++) {
+            corrected_location[i] = state[i].location;
+            location_offset(corrected_location[i], _NE_pos_offset_m[i].x, _NE_pos_offset_m[i].y);
+            corrected_location[i].alt += (int)(_hgt_offset_cm[i]);
+    }
+
+    // Calculate the weighted sum of horizontal and vertical position offsets relative to the blended location
+    blended_alt_offset_cm = 0.0f;
+    blended_NE_offset_m.zero();
+    for (uint8_t i=0; i<GPS_MAX_RECEIVERS; i++) {
+        if (_blend_weights[i] > 0.0f) {
+            blended_NE_offset_m += location_diff(state[GPS_MAX_RECEIVERS].location, corrected_location[i]) * _blend_weights[i];
+            blended_alt_offset_cm += (float)(corrected_location[i].alt - state[GPS_MAX_RECEIVERS].location.alt) * _blend_weights[i];
+        }
+    }
+
+    // If the GPS week is the same then use a blended time_week_ms
+    // If week is different, then use time stamp from GPs with largest weighting
+    // detect inconsistent week data
+    uint8_t last_week_instance = 0;
+    bool weeks_inconsistent = false;
+    for (uint8_t i=0; i<GPS_MAX_RECEIVERS; i++) {
+        if (last_week_instance == 0 && _blend_weights[i] > 0) {
+            // this is our first valid sensor week data
+            last_week_instance = state[i].time_week;
+        } else if (last_week_instance != 0 && _blend_weights[i] > 0 && last_week_instance != state[i].time_week) {
+            // there is valid sensor week data that is inconsistent
+            weeks_inconsistent = true;
+        }
+    }
+    // calculate output
+    if (!weeks_inconsistent) {
+        // use data from highest weighted sensor
+        state[GPS_MAX_RECEIVERS].time_week = state[best_index].time_week;
+        state[GPS_MAX_RECEIVERS].time_week_ms = state[best_index].time_week_ms;
+    } else {
+        // use week number from highest weighting GPS (they should all have the same week number)
+        state[GPS_MAX_RECEIVERS].time_week = state[best_index].time_week;
+        // calculate a blended value for the number of ms lapsed in the week
+        double temp_time_0 = 0.0;
+        for (uint8_t i=0; i<GPS_MAX_RECEIVERS; i++) {
+            if (_blend_weights[i] > 0.0f) {
+                temp_time_0 += (double)state[i].time_week_ms * _blend_weights[i];
+            }
+        }
+        state[GPS_MAX_RECEIVERS].time_week_ms = (uint32_t)temp_time_0;
+    }
+
+    // calculate a blended value for the timing data
+    double temp_time_1 = 0.0;
+    double temp_time_2 = 0.0;
+    for (uint8_t i=0; i<GPS_MAX_RECEIVERS; i++) {
+        if (_blend_weights[i] > 0.0f) {
+            temp_time_1 += (double)timing[i].last_fix_time_ms * _blend_weights[i];
+            temp_time_2 += (double)timing[i].last_message_time_ms * _blend_weights[i];
+        }
+    }
+    timing[GPS_MAX_RECEIVERS].last_fix_time_ms = (uint32_t)temp_time_1;
+    timing[GPS_MAX_RECEIVERS].last_message_time_ms = (uint32_t)temp_time_2;
```

Comment:
If we are blending these here there's no need to be setting them to the newest value above.